### ISSUE TITLE
[PUI] Purchase order total cost not updated when adding extra line items

### ISSUE THREAD

**Issue by sur5r** [inventree/InvenTree#8695]
### Please verify that this bug has NOT been raised before.

- [x] I checked and didn't find a similar issue

### Describe the bug*

When adding an extra line item to a purchase order, the total cost is not updated unless the pages is reloaded.

### Steps to Reproduce

1. Go to a purchase order
2. Add an extra line item with a price
3. Click on "Order details"
4. Observe the total cost not including the price of the extra line item

### Expected behaviour

The total cost in "Order details" gets updated without a page reload.

### Deployment Method

- [ ] Docker
- [ ] Package
- [x] Bare metal
- [ ] Other - added info in Steps to Reproduce

### Version Information

InvenTree-Version: 0.18.0 dev
Django Version: 4.2.17
Commit Hash: ff69cf6
Commit Date: 2024-12-17
Commit Branch: master
Database: postgresql
Debug-Mode: False
Deployed using Docker: False
Platform: Linux-6.1.0-13-cloud-amd64-x86_64-with-glibc2.36
Installer: GIT
nullActive plugins: [{"name":"InvenTreeBarcode","slug":"inventreebarcode","version":"2.1.0"},{"name":"InvenTreeCoreNotificationsPlugin","slug":"inventreecorenotificationsplugin","version":"1.0.0"},{"name":"InvenTreeCurrencyExchange","slug":"inventreecurrencyexchange","version":"1.0.0"},{"name":"InvenTreeLabel","slug":"inventreelabel","version":"1.1.0"},{"name":"InvenTreeLabelMachine","slug":"inventreelabelmachine","version":"1.0.0"},{"name":"InvenTreeLabelSheet","slug":"inventreelabelsheet","version":"1.0.0"},{"name":"DigiKeyPlugin","slug":"digikeyplugin","version":"1.0.0"},{"name":"LCSCPlugin","slug":"lcscplugin","version":"1.0.0"},{"name":"MouserPlugin","slug":"mouserplugin","version":"1.0.0"},{"name":"TMEPlugin","slug":"tmeplugin","version":"1.0.0"},{"name":"Brother Labels","slug":"brother","version":"1.0.0"}]

### Please verify if you can reproduce this bug on the demo site.

- [x] I can reproduce this bug on the demo site.

### Relevant log output

```shell

```

**Comment by sur5r** [comment]
No, CUI is gone as I'm on current master. I just reproduced in on the demo instance, I can create screenshots there if wanted.

**Comment by SchrodingersGat** [comment]
@sur5r sorry this one has been buried - if you can reproduce this still and provide screenshots that would be appreciated.

**Comment by sur5r** [comment]
Will do, but it might take until Wednesday until I get to it.

**Comment by sur5r** [comment]
Reproduction steps with screenshots:

1. Go to https://demo.inventree.org/web/purchasing/purchase-order/19/detail
![Image](https://github.com/user-attachments/assets/92641706-b76c-4b0d-ad6c-bb499c19a2aa)
Notice total cost of $0, 0 line items.
2. Go to "Line items"
![Image](https://github.com/user-attachments/assets/02bee541-059b-42da-8f6d-fcb1bae2e4b7)
3. Add a line items
![Image](https://github.com/user-attachments/assets/bbea929a-bfda-46d8-b8e8-4ed836cfe2dd)
4. Line item and price is shown
![Image](https://github.com/user-attachments/assets/7833d89e-a04a-441b-ba55-c053f5484d31)
5. Click on "Order details"
![Image](https://github.com/user-attachments/assets/71fbb4b8-9796-43f8-8030-0bd55223e602)
Notice total cost is still at $0, same for line item count.
6. Reload the page (F5)
![Image](https://github.com/user-attachments/assets/96dd4bf0-78d3-4fbd-ba98-2bec8a5a7a6f)
Now the summary data is correct.

**Comment by SchrodingersGat** [comment]
Ah, thanks for the detail. So, we need to trigger a refresh of the order details whenever a line item is added / edited / deleted

**Comment by SchrodingersGat** [comment]
@sur5r is this something you would be interested in submitting a fix for?

**Comment by sur5r** [comment]
I will give it a try. I think I found the necessary bits already.
